File: src/python/PythonSDK/foundationallm/langchain/data_sources/search_service/search_service_configuration.py
"""
Class name: SearchServiceConfiguration
Description: Azure AI Search index information.
"""
from typing import Optional, Literal
from foundationallm.langchain.data_sources import DataSourceConfiguration

class SearchServiceConfiguration(DataSourceConfiguration):
    """
    Connection information indicating the index information
    and the number of results to return in the RAG pattern
    """
    configuration_type: Literal['search_service']
    endpoint: str
    key_secret: str
    index_name: str
    embedding_field_name: Optional[str] = "Embedding"
    text_field_name: Optional[str] = "Text"
    top_n: Optional[int] = 3

File: src/dotnet/AgentFactory/Models/Orchestration/DataSourceConfigurations/SearchServiceConfiguration.cs
using Newtonsoft.Json;

namespace FoundationaLLM.AgentFactory.Core.Models.Orchestration.DataSourceConfigurations
{
    /// <summary>
    /// Search Service configuration settings.
    /// </summary>
    public class SearchServiceConfiguration
    {
        /// <summary>
        /// The type of configuration. This value should not be changed.
        /// </summary>
        [JsonProperty("configuration_type")]
        public string ConfigurationType = "search_service";

        /// <summary>
        /// The endpoint of the Search Service
        /// </summary>
        [JsonProperty("endpoint")]
        public string? Endpoint { get; set; }

        /// <summary>
        /// The name of key vault secret containing the access key
        /// </summary>
        [JsonProperty("key_secret")]
        public string? KeySecret { get; set; }

        /// <summary>
        /// The name of the index in Azure AI Search
        /// </summary>
        [JsonProperty("index_name")]
        public string? IndexName { get; set; }

        /// <summary>
        /// (Optional) The number of rows to return from the index search
        /// </summary>
        [JsonProperty("top_n")]
        public int? TopN { get; set; }

        /// <summary>
        /// (Optional) The name of the field to use for embedding in the Search Service.
        /// </summary>
        [JsonProperty("embedding_field_name")]
        public string? EmbeddingFieldName { get; set; }

        /// <summary>
        /// (Optional) The name of the field to use for raw text in the Search Service.
        /// </summary>
        [JsonProperty("text_field_name")]
        public string? TextFieldName { get; set; }
    }
}
